### PR TITLE
Updated to eslint.

### PR DESCRIPTION
**Pull request by Samsabal**
Made use of the @typescript-eslint/eslint-plugin-tslint package because we where already using tslint.

**Note:**
There rules maybe need to be adjusted to the projects needs.

Fixes #22

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+module.exports = {
+    'root': true,
+    'env': {
+        'node': true
+    },
+    'extends': [
+        'eslint:recommended',
+        'plugin:unicorn/recommended',
```
Comment (SleeplessByte) [inline .eslintrc.js]: What is this unicorn plugin? It does not seem to be present in the original tslint.json so I don't understand why it's present here.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        'ecmaVersion': 2017,
+        'sourceType': 'module',
+        'ecmaFeatures': {
+            'jsx': false,
+        },
+        'project': 'tsconfig.json',
+    },
+    'plugins': [
+        'unicorn',
+        'import',
```
Comment (SleeplessByte) [inline .eslintrc.js]: Same as above

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    'parserOptions': {
+        'ecmaVersion': 2017,
+        'sourceType': 'module',
+        'ecmaFeatures': {
+            'jsx': false,
+        },
+        'project': 'tsconfig.json',
+    },
+    'plugins': [
+        'unicorn',
```
Comment (SleeplessByte) [inline .eslintrc.js]: Same as above

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        'import',
+        '@typescript-eslint/tslint',
```
Comment (SleeplessByte) [inline .eslintrc.js]: According to the [documentation](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin-tslint) of [`@typescript-eslint/tslint`](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin-tslint) this runs `tslint` on top of `eslint`. That is _not_ what we want. We want to _remove_ `tslint` completely. 

Use [`@typescript-eslint`](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin) instead. It's more work than just "copy paste".

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    ],
+    'rules': {
```
Comment (SleeplessByte) [inline .eslintrc.js]: Are you sure these are the rules currently in tslint? Please elaborate on why these rules need to be specified here.

File: package.json [hunk 6]
```diff
@@ -43,5 +43,8 @@
     "require": [
       "esm"
     ]
+  },
+  "dependencies": {
+    "@typescript-eslint/eslint-plugin-tslint": "^1.6.0"
```
Comment (SleeplessByte) [inline package.json]: We don't want this one, see above. Also, this should be a `devDependency`.

Missing the import and unicorn plugin dependency here. Might not need them at all.

Comment (SleeplessByte) [reply package.json]: Still has `tslint` in `devDependencies`.

Comment (SleeplessByte) [reply package.json]: We need `prettier` to work with `eslint`.

File: .eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    ],
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+    "ecmaFeatures": {
+      "jsx": true
+    },
+    "useJSXTextNode": true,
+    "project": "./tsconfig.json",
+    "tsconfigRootDir": "../../",
+    "extraFileExtensions": [".vue"]
```
Comment (SleeplessByte) [inline .eslintrc.js]: I don't think this uses vue

File: .eslintrc.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+module.exports = {
+    'extends': [
+        "plugin:@typescript-eslint/recommended",
+        "prettier",
+        "prettier/@typescript-eslint",
+    ],
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+    "ecmaFeatures": {
+      "jsx": true
```
Comment (SleeplessByte) [inline .eslintrc.js]: I don't think this project uses JSX

File: .eslintrc.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "useJSXTextNode": true,
```
Comment (SleeplessByte) [inline .eslintrc.js]: I don't think this project uses JSX

File: .eslintrc.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    "project": "./tsconfig.json",
+    "tsconfigRootDir": "../../",
```
Comment (SleeplessByte) [inline .eslintrc.js]: Are you certain? I don't tink this is right.

File: .prettierrc.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
+module.exports =  {
+    semi:  true,
+    trailingComma:  'all',
+    singleQuote:  true,
+    printWidth:  120,
+    tabWidth:  4,
```
Comment (SleeplessByte) [inline .prettierrc.js]: This project was 2 spaces. And it should remain.

File: package.json [hunk 12]
```diff
@@ -25,18 +25,22 @@
...
     "ava": "^1.4.1",
+    "eslint": "^5.16.0",
+    "eslint-plugin-import": "^2.17.2",
+    "eslint-config-prettier": "^4.1.0",
+    "eslint-plugin-prettier": "^3.0.1",
     "esm": "^3.2.22",
     "microbundle": "^0.11.0",
     "nyc": "^13.3.0",
-    "prettier": "1.16.4",
+    "prettier": "^1.17.0",
```
Comment (SleeplessByte) [inline package.json]: Don't use a `^` version for prettier but lock it add an `--exact` version.

File: moxie.ts [hunk 13]
```diff
@@ -263,9 +236,7 @@ const handler = {
     }
 
     const expectedCalls = Object.keys(mock.expectedCalls) || ['<nothing>']
-    throw new ArgumentError(
-      `unmocked method ${name}, expected one of ${mock.__print(expectedCalls)}`
-    )
+    throw new ArgumentError(`unmocked method ${name}, expected one of ${mock.__print(expectedCalls)}`)
```
Comment (SleeplessByte) [inline moxie.ts]: This looks like prettier is not correctly running?

**Comment by SleeplessByte** [comment]
@Samsabal

File: .prettierrc.js [hunk 14]
```diff
@@ -0,0 +1,8 @@
+module.exports =  {
+    semi:  false,
+    trailingComma:  "none",
+    singleQuote:  true,
+    printWidth:  120,
```
Comment (SleeplessByte) [inline .prettierrc.js]: I think it was 80?

Comment (Samsabal) [reply .prettierrc.js]: Can't really find where it was 80 in the file history. But the default value is 80 so I will change it anyways. 👌

Comment (SleeplessByte) [reply .prettierrc.js]: Yes, it was not set, so it was the default.

**Comment by SleeplessByte** [comment]
I don't know why, but you removed the `format` script, so CI is failing
https://travis-ci.com/XPBytes/moxie/jobs/205252279

**Pull request merged**